common/build-style/gem.sh

```diff
@@ -0,0 +1,125 @@
+#
+# This helper is for templates using gems from Rubygems or gemspec files.
+#
+do_build() {
+	: ${gem_cmd:=gem}
+
+	# Some packages keep Gem.gemspec as the name instead of the proper
+	# $pkgname.gemspec, fix those
+	if [ -f Gem.gemspec ]; then
+		mv Gem.gemspec ${pkgname#ruby-}.gemspec
+	fi
+
+	# Replace name and version just in case they try something funny like
+	# requring RELEASE to be on the environment to not append -rc0 to version
+	# Some even forget to update the version in the gemspec after releasing
+	sed -ri "s|(\.name .*)=.*|\1 = \"${pkgname#ruby-}\"|g" ${pkgname#ruby-}.gemspec
+	sed -ri "s|(\.version .*)=.*|\1 = \"${version}\"|g" ${pkgname#ruby-}.gemspec
+
+	# Replace use of `git ls-files` with `find . -type f`
+	sed -i 's|`git ls-files`|`find . -type f`|g' ${pkgname#ruby-}.gemspec
+
+	# Same as above but use -print0 when people call git ls-files -z
+	sed -i 's|`git ls-files -z`|`find . -type f`|g' ${pkgname#ruby-}.gemspec
+
+	if [ "$CROSS_BUILD" ]; then
+		# Create a .extconf file that forces the Makefile to use our environment
+		# this allows us to cross-compile like it is done with meson cross-files
+		cat>append<<EOF
+RbConfig::MAKEFILE_CONFIG['CPPFLAGS'] = ENV['CPPFLAGS'] if ENV['CPPFLAGS']
+\$CPPFLAGS = ENV['CPPFLAGS'] if ENV['CPPFLAGS']
+RbConfig::MAKEFILE_CONFIG['CC'] = ENV['CC'] if ENV['CC']
+RbConfig::MAKEFILE_CONFIG['CXX'] = ENV['CXX'] if ENV['CXX']
+RbConfig::MAKEFILE_CONFIG['LD'] = ENV['LD'] if ENV['LD']
+RbConfig::MAKEFILE_CONFIG['CFLAGS'] = ENV['CFLAGS'] if ENV['CFLAGS']
+RbConfig::MAKEFILE_CONFIG['CXXFLAGS'] = ENV['CXXFLAGS'] if ENV['CXXFLAGS']
+EOF
+
+		# Patch all instances of extconf that use create_makefile
+		for f in $(find . -type f -name 'extconf.rb'); do
+			if [ ! -f ${f}.orig ]; then
+				# Ignore extconf files that do not create makefiles
+				grep -q create_makefile $f || continue
+				# Create a backup which we will restore later
+				cp $f ${f}.orig
+				# Patch extconf.rb for cross compile
+				cat append ${f}.orig >> $f
+			fi
+		done
```

Comment:
Hm, maybe doing `patch` and later on `patch -R` would be a little more elegant?

Reply:
Just write the stuff you write into append into a patch file, do `patch -i $patch_file` when you want to modify it and `patch -R -i $patch_file` when you want to revert it.